### PR TITLE
SDIT-1378 Provide Incident Reconciliation counts

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/config/CoroutineScopes.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package uk.gov.justice.digital.hmpps.prisonerfromnomismigration.config
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/config/CoroutineScopes.kt:1]: Pulled in from prisoner-to-nomis-update

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/service/HelperApi.kt [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package uk.gov.justice.digital.hmpps.prisonerfromnomismigration.service
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/service/HelperApi.kt:1]: Pulled in from prisoner-to-nomis-update

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incidents/IncidentsReconciliationService.kt [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+class IncidentsReconciliationService(
+  private val telemetryClient: TelemetryClient,
+  private val incidentsService: IncidentsService,
+  private val nomisIncidentsApiService: IncidentsNomisApiService,
+) {
+  private companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  suspend fun generateReconciliationReport(agencies: List<IncidentAgencyId>): List<MismatchIncidents> =
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incidents/IncidentsReconciliationService.kt:25]: Wasn't sure if the List of incidentAgencyId should be paged...there will be just over 120 ish returned

Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incidents/IncidentsReconciliationService.kt:25]: nah, not worth it

**Review by andymarke — APPROVED** [review]
lgtm

**Pull request merged**